spec/serializers/post_serializer_spec.rb

```diff
@@ -438,6 +438,71 @@ def json_for_user(user)
     end
   end
 
+  describe "#badges_granted" do
+    fab!(:user)
+    fab!(:post) { Fabricate(:post, user: user) }
+    fab!(:badge1) do
+      Badge.create!(
+        name: "SomeBadge",
+        badge_type_id: BadgeType::Bronze,
+        show_posts: true,
+        post_header: true,
+      )
+    end
+    fab!(:badge2) do
+      Badge.create!(
+        name: "AnotherBadge",
+        badge_type_id: BadgeType::Bronze,
+        show_posts: true,
+        post_header: true,
+      )
+    end
+    fab!(:ub1) do
+      UserBadge.create!(
+        badge_id: badge1.id,
+        user: user,
+        granted_by: Discourse.system_user,
+        granted_at: Time.now,
+        post_id: post.id,
+      )
+    end
+    fab!(:ub2) do
+      UserBadge.create!(
+        badge_id: badge2.id,
+        user: user,
+        granted_by: Discourse.system_user,
+        granted_at: Time.now,
+        post_id: post.id,
+      )
+    end
+
+    let(:serializer) { described_class.new(post, scope: Guardian.new(user), root: false) }
+
+    it "doesn't include badges when disabled" do
```

Comment:
```suggestion
    it "doesn't include badges when `enable_badges` site setting is disabled" do
```

I find `when disabled` alittle vague because it doesn't specify what is actually being disabled.